arelle/RuntimeOptions.py

```diff
@@ -144,12 +146,35 @@ class RuntimeOptions:
     webserver: Optional[str] = None
     xdgConfigHome: Optional[str] = None
 
+    def __delattr__(self, name: str) -> Any:
+        """
+        Delete attribute while silencing AttributeError if it does not exist and `strictOptions` is False.
+        :param name:
+        :return: None
+        """
+        try:
+            object.__delattr__(self, name)
+        except AttributeError as exc:
+            if self.strictOptions:
+                raise exc
+
     def __eq__(self, other: Any) -> bool:
         """ Default dataclass implementation doesn't consider plugin applied options. """
         if isinstance(other, RuntimeOptions):
             return vars(self) == vars(other)
         return NotImplemented
 
+    def __getattr__(self, name: str) -> Any:
+        """
+        If an attribute isn't found, it may be an unspecified plugin option.
+        Return None for any attributes not found on the class.
+        :param name:
+        :return: None
+        """
+        if self._initialized and not self.strictOptions:
+            return None
+        raise AttributeError(name)
```

Comment:
Ideally users of the Python API would not have to specify default values or `None` for every attribute that a plugin might use, and plugins themselves wouldn't have to make use of `hasattr` to determine if an attribute was provided. 

Setting `strictOptions=False` allows attributes like `viewerURL` in `ixbrl-viewer`, for example, to be unset in `RuntimeOptions` while allowing the plugin to reference the attribute and know that it will either contain a value or be `None` and will never raise an `AttributeError`.